frontend/front/src/pages/Public/Components/LangPrefDropdown.js

```diff
@@ -0,0 +1,144 @@
+import React, { useState, useEffect } from "react";
+import { useLocation } from "react-router-dom";
+import { Box, Menu, MenuItem, Fade, Button, Typography } from "@mui/material";
+import LanguageIcon from "@mui/icons-material/Language";
+import KeyboardArrowDownIcon from "@mui/icons-material/KeyboardArrowDown";
+import { useTranslation } from "react-i18next";
+
+const LangPrefDropdown = () => {
+  const {
+    t,
+    i18n: { changeLanguage, language },
+  } = useTranslation();
+
+  const [anchorMore, setAnchorMore] = useState(null);
+  const [langDisplay, setLangDisplay] = useState("English");
+  const [currentLanguage, setCurrentLanguage] = useState(language);
+
+  const location = useLocation();
+
+  const langMap = {
+    "en-us": `🇺🇸 ${t("public.global-labels.locales.english")}`,
+    "ht-cr": `🇭🇹 ${t("public.global-labels.locales.creole")}`,
+    "pt-br": `🇧🇷 ${t("public.global-labels.locales.portuguese")}`,
+    "es-us": `🇪🇸 ${t("public.global-labels.locales.spanish")}`,
+  };
+
+  useEffect(() => {
+    // Get query params from current URL
+    const params = new URLSearchParams(window.location.search);
+    let queryLang = params.get("langPref");
+    // Check if current route is a 'public' route
+    const isPublicRoute = window.location.pathname.includes("public");
+
+    if (isPublicRoute) {
+      // Get language preference from localStorage or default to 'en-us'
+      queryLang = localStorage.getItem("langPref") || "en-us";
```

Comment:
What is the difference between `langPref` and `i18nextLng` in local storage? For me, they are always identical.
![image](https://github.com/codeforboston/urban-league-heat-pump-accelerator/assets/93659375/7721f55c-9bc2-46cb-ae22-f63f4c4a2174)

Reply:
i18nextLng is just for debugging purposes. I'll set debug: true to false in i18n.js, so it won't show up anymore.